includes/class-woothemes-sensei-quiz.php

```diff
@@ -1053,4 +1056,35 @@ public function get_user_question_feedback( $lesson_id, $question_id, $user_id =
 
      } // end get_user_question_feedback
 
+     /**
+      * Check if a quiz has no questions, and redirect back to lesson.
+      *
+      * Though a quiz is created for each lesson, it should not be visible
+      * unless it has questions.
+      *
+      * @since 1.9.0
+      * @access public
+      * @param none
+      * @return void
+      */
+
+     public function quiz_has_no_questions() {
+
+         global $post;
+
+         $lesson_id = $this->get_lesson_id($post->ID);
+
+         $has_questions = get_post_meta( $lesson_id, '_quiz_has_questions', true );
```

Comment:
Hi @dwainm, @mahangu, every time a quiz is updated within the admin it calls quiz_update() on the Lesson class. Within that are the lines https://github.com/woothemes/sensei/blob/master/includes/class-woothemes-sensei-lesson.php#L427-L433 which do the counting and setting/removing of the '_quiz_has_questions' meta key. As far as i'm aware, there's no other way to attach questions to a quiz. Getting all quiz questions and counting every time to see if there are questions is quite the overhead.